src/sidebar/util/time.ts

```diff
@@ -262,3 +263,30 @@ export function formatDate(date: Date, Intl?: IntlType): string {
     Intl,
   );
 }
+
+/**
+ * Formats a date as `YYYY-MM-DD hh:mm`, using 24h.
+ * @param timeZone Defaults to system date when not provided
+ */
+export function formatDateTime(date: Date, timeZone?: string): string {
+  const options: Intl.DateTimeFormatOptions = {
+    year: 'numeric',
+    month: '2-digit',
+    day: '2-digit',
+    hour: '2-digit',
+    minute: '2-digit',
+    second: '2-digit',
+    hour12: false,
+    timeZone,
+  };
+
+  // Format the date for a fixed locale (en-GB) where we know the output and
+  // how to parse it afterward.
+  // Here we know formattedDate will have "DD/MM/YYYY, hh:mm:ss" format.
```

Comment:
I couldn't be certain that a fixed locale will produce the same format in every browser and over time. If we're not actually using any aspect of `Intl` formatting, we could just construct the format string entirely ourselves. We will just need to take care that the timezone is handled as intended.

I looked around to see if there was a standard format / specification for the format we are constructing, that we could use as a reference. It looks like Postgres's time stamps (and the same or very similar formats are common in SQL generally) are a good match: https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-DATETIME-INPUT-TIME-STAMPS.

Reply:
> I couldn't be certain that a fixed locale will produce the same format in every browser and over time. If we're not actually using any aspect of Intl formatting, we could just construct the format string entirely ourselves. We will just need to take care that the timezone is handled as intended.

Yeah, I thought about this risk. However, I couldn't find at first another way to reliably generate the pieces properly for a specific timezone (which at runtime will be the system/browser timezone).

By doing another search with the learnings from this, I see now at least one other option which could potentially work.

```ts
const dateStringInAnotherTimezone = date.toLocaleString('en-US', { timeZone: 'America/New_York' });
const newDateInAnotherTimezone = new Date(dateStringInAnotherTimezone);

// Then get every individual component that we need from newDateInAnotherTimezone
```

This seems to work, although I need to check if it works when timeZone is not explicitly provided, in falling back to system/browser timezone.

It still relies in formatting for one locale at one point, but then we feed that back to the internal browser APIs, so we don't directly rely on that.

I'll rewrite it and see if tests keep passing.

Reply:
>  I need to check if it works when timeZone is not explicitly provided, in falling back to system/browser timezone

This works as well